core/src/services/tikv/backend.rs

```diff
@@ -0,0 +1,244 @@
+    pub fn endpoints(&mut self, endpoints: Vec<String>) -> &mut Self {
+        let ep: Vec<String> = endpoints.into_iter().collect();
+        if !ep.is_empty() {
+            self.endpoints = Some(ep)
+        }
+        self
+    }
+
+    /// Set the insecure connection to TiKV.
+    pub fn insecure(&mut self) -> &mut Self {
+        self.insecure = false;
+        self
+    }
+
+    /// Set the certificate authority file path.
+    pub fn ca_path(&mut self, ca_path: &str) -> &mut Self {
+        if !ca_path.is_empty() {
+            self.ca_path = Some(ca_path.to_string())
+        }
+        self
+    }
+
+    /// Set the certificate file path.
+    pub fn cert_path(&mut self, cert_path: &str) -> &mut Self {
+        if !cert_path.is_empty() {
+            self.cert_path = Some(cert_path.to_string())
+        }
+        self
+    }
+
+    /// Set the key file path.
+    pub fn key_path(&mut self, key_path: &str) -> &mut Self {
+        if !key_path.is_empty() {
+            self.key_path = Some(key_path.to_string())
+        }
+        self
+    }
+}
+
+impl Builder for TikvBuilder {
+    const SCHEME: Scheme = Scheme::Redb;
+    type Accessor = Backend;
+
+    fn from_map(map: HashMap<String, String>) -> Self {
+        let mut builder = TikvBuilder::default();
+
+        map.get("endpoints")
+            .map(|v| v.split(',').map(|s| s.to_owned()).collect::<Vec<String>>())
+            .map(|v| builder.endpoints(v));
+        map.get("insecure")
+            .filter(|v| *v == "on" || *v == "true")
+            .map(|_| builder.insecure());
+        map.get("ca_path").map(|v| builder.ca_path(v));
+        map.get("cert_path").map(|v| builder.cert_path(v));
+        map.get("key_path").map(|v| builder.key_path(v));
+
+        builder
+    }
+
+    fn build(&mut self) -> Result<Self::Accessor> {
+        let endpoints = self.endpoints.take().ok_or_else(|| {
+            Error::new(
+                ErrorKind::ConfigInvalid,
+                "endpoints is required but not set",
+            )
+            .with_context("service", Scheme::Tikv)
+        })?;
+
+        if self.insecure
+            && (self.ca_path.is_some() || self.key_path.is_some() || self.cert_path.is_some())
+        {
+            return Err(
+                Error::new(ErrorKind::ConfigInvalid, "invalid tls configuration")
+                    .with_context("service", Scheme::Tikv)
+                    .with_context("endpoints", format!("{:?}", endpoints)),
+            )?;
+        }
+
+        Ok(Backend::new(Adapter {
+            client: OnceCell::new(),
+            endpoints,
+            insecure: self.insecure,
+            ca_path: self.ca_path.clone(),
+            cert_path: self.cert_path.clone(),
+            key_path: self.key_path.clone(),
+        }))
+    }
+}
+
+/// Backend for TiKV service
+pub type Backend = kv::Backend<Adapter>;
+
+#[derive(Clone)]
+pub struct Adapter {
+    client: OnceCell<RawClient>,
```

Comment:
It's better to support both `TxnClient` and `RawClient`, but it's ok to leave to future PRs.

Reply:
I can not get this, In our use case, we do not need to use transactions to do this (exclude `scan`)?

Reply:
> I can not get this, In our use case, we do not need to use transactions to do this (exclude `scan`)?

Users could write data into TiKV in other ways, we need to make sure we didn't read/write    data unexpectedly.

Reply:
So you mean we need to use `TxnClient` to replace `RawClient`?

Reply:
> So you mean we need to use `TxnClient` to replace `RawClient`?

My plan is to support both `TxnClient` and `RawClient`, allowing our users to choose which one they prefer.